src/scenic/core/object_types.py

```diff
@@ -1769,11 +1769,11 @@ def __init_subclass__(cls):
             cls._props_transformed = str(cls)
 
             props = cls._scenic_properties
-            # Raise error if parentOrientation already defined
-            if "parentOrientation" in props:
+            # Raise error if parentOrientation and heading already defined
+            if "parentOrientation" in props and "heading" in props:
                 raise RuntimeError(
                     "this scenario cannot be run with the --2d flag (the "
-                    f'{cls.__name__} class defines "parentOrientation")'
+                    f'{cls.__name__} class defines "parentOrientation" and "heading")'
```

Comment:
Hmm, I think still saying `this scenario cannot be run with the --2d flag` is misleading, since the scenario won't work without that flag either (you'd get a `"heading" property cannot be overridden` error). How about `<name> defines both "parentOrientation" and "heading"`?